### PR TITLE
707 implement the mcmc prob dose fit methods for the logisticlognormalordinal class

### PR DESCRIPTION
**Pull request by Puzzled-Face**

### CODE REVIEW

File: R/Model-class.R [hunk 1]
```diff
@@ -3335,3 +3335,98 @@ FractionalCRM <- function(...) {
...
+#' using an ordinal toxicity scale.
+#'
+#' @aliases LogisticLogNormalOrdinal
+#' @export
+.LogisticLogNormalOrdinal <- setClass(
+  Class = "LogisticLogNormalOrdinal",
+  contains = "ModelLogNormal",
+  slots = c(datablock = "function")
+  # ,
+  # validity = v_model_ordinal_logistic_log_normal
```
Comment (danielinteractive) [inline R/Model-class.R:3355]: Would be good to have the validation function?

File: R/Model-class.R [hunk 2]
```diff
@@ -3335,3 +3335,98 @@ FractionalCRM <- function(...) {
...
+#' @export
+.DefaultLogisticLogNormalOrdinal <- function() {
+  LogisticLogNormalOrdinal(
+    mean = c(-3, -4, 1),
+    cov = diag(c(3, 4, 1)),
+    ref_dose = 50
+  )
+}
+
+ordinal_data <- DataOrdinal(
```
Comment (danielinteractive) [inline R/Model-class.R:3425]: This should not be here, I guess needs to go to example file

File: R/Samples-methods.R [hunk 3]
```diff
@@ -921,6 +921,71 @@ setMethod("fit",
       return(ret)
     }
 )
+
+##' @param points at which dose levels is the fit requested? default is the dose
```
Comment (danielinteractive) [inline R/Samples-methods.R:925]:
```suggestion
#' @param points at which dose levels is the fit requested? default is the dose
```
Etc below

File: R/Samples-methods.R [hunk 4]
```diff
@@ -921,6 +921,71 @@ setMethod("fit",
...
+##' grid
+##' @param quantiles the quantiles to be calculated (default: 0.025 and
```
Comment (danielinteractive) [inline R/Samples-methods.R:927]: Please don't repeat generic function argument description

**Review by danielinteractive — COMMENTED** [review]
Thanks @Puzzled-Face looks nice, please see below comments

File: R/Model-class.R (hunk 1 above)
Comment (Puzzled-Face) [reply R/Model-class.R]: The `datablock` slot is a left over from a previous iteration of the design.  It's not necessary and I have deleted it.  Therefore, I think there is no need for a specific validation function for `LogisticLogNormalOrdinal`.  Do you agree?

File: R/Model-class.R (hunk 2 above)
Comment (Puzzled-Face) [reply R/Model-class.R:3422]: Yep.  My bad.

File: R/Samples-methods.R (hunk 3 above)
Comment (Puzzled-Face) [reply R/Samples-methods.R:925]: Already fixed.

File: R/Model-class.R (hunk 1 above)
Comment (danielinteractive) [reply R/Model-class.R]: yep sounds good - then just delete the commented code

**Review by danielinteractive — APPROVED** [review]
Thanks @Puzzled-Face !

**Pull request merged**